parsers/html.c

```diff
@@ -8,35 +8,393 @@
+		*lineOffset = getInputLineOffset ();
+		readToken (token, FALSE);
+		type = token->type;
+		if (type == TOKEN_TAG_START)
+			readTag (token, text);
+		if (type == TOKEN_COMMENT || type == TOKEN_TAG_START)
+		{
+			readTokenText (token, text != NULL);
+			if (text != NULL)
+				vStringCat (text, token->string);
+		}
+	}
+	while (type == TOKEN_COMMENT || type == TOKEN_TAG_START);
+
+	return type == TOKEN_TAG_START2;
+}
+
+static void readTag (tokenInfo *token, vString *text)
+{
+	boolean textCreated = FALSE;
+
+	readToken (token, TRUE);
+	if (token->type == TOKEN_NAME)
+	{
+		keywordId startTag;
+		boolean isHeading;
+
+		startTag = lookupKeyword (vStringValue (token->string), Lang_html);
+		isHeading = (startTag == KEYWORD_h1 || startTag == KEYWORD_h2 || startTag == KEYWORD_h3);
+		if (text == NULL && isHeading)
+		{
+			text = vStringNew ();
+			textCreated = TRUE;
+		}
+
+		do
+		{
+			readToken (token, TRUE);
+			if (startTag == KEYWORD_a && token->type == TOKEN_NAME)
+			{
+				keywordId attribute = lookupKeyword (vStringValue (token->string), Lang_html);
+
+				if (attribute == KEYWORD_name)
+				{
+					readToken (token, TRUE);
+					if (token->type == TOKEN_EQUAL)
+					{
+						readToken (token, TRUE);
+						if (token->type == TOKEN_STRING || token->type == TOKEN_NAME)
+							makeSimpleTag (token->string, HtmlKinds, K_ANCHOR);
+					}
+				}
+			}
+		}
+		while (token->type != TOKEN_TAG_END && token->type != TOKEN_TAG_END2 &&
+			   token->type != TOKEN_EOF);
+
+		if (token->type == TOKEN_TAG_END)
+		{
+			long startSourceLineNumber = getSourceLineNumber ();
+			long startLineNumber = getInputLineNumber ();
+			int startLineOffset = getInputLineOffset ();
+			long endLineNumber;
+			int endLineOffset;
+			boolean tag_start2;
+
+			tag_start2 = readTagContent (token, text, &endLineNumber, &endLineOffset);
+
+			if (tag_start2)
+			{
+				readToken (token, TRUE);
+				if (isHeading && textCreated && vStringLength (text) > 0)
+				{
+					keywordId endTag = lookupKeyword (vStringValue (token->string), Lang_html);
+					if (startTag == endTag)
+					{
+						htmlKind headingKind;
+
+						if (startTag == KEYWORD_h1)
+							headingKind = K_HEADING1;
+						else if (startTag == KEYWORD_h2)
+							headingKind = K_HEADING2;
+						else
+							headingKind = K_HEADING3;
+
+						makeSimpleTag (text, HtmlKinds, headingKind);
+					}
+				}
+				else if (startTag == KEYWORD_script)
+				{
+					keywordId endTag = lookupKeyword (vStringValue (token->string), Lang_html);
+					if (startTag == endTag)
+						makePromise ("JavaScript", startLineNumber, startLineOffset,
+									 endLineNumber, endLineOffset, startSourceLineNumber);
+				}
+				else if (startTag == KEYWORD_style)
+				{
+					keywordId endTag = lookupKeyword (vStringValue (token->string), Lang_html);
+					if (startTag == endTag)
+						makePromise ("CSS", startLineNumber, startLineOffset,
+									 endLineNumber, endLineOffset, startSourceLineNumber);
```

Comment:
That they [can technically be something else than CSS](https://www.w3.org/TR/html5/document-metadata.html#attr-style-type), but again, forget that, nobody actually use anything else.

But well, if we wanted to be totally strict, we'd have to check whether the `type` attribute is provided and then if it's the CSS MIME type.